In this project, we will build a powerful yet intuitive toolbox to support and automate discovery of complex prognostic biomarkers from transcriptomic data. 

Multigene biomarkers of disease are able to capture patients' physiological state at the point of diagnosis or treatment decision, and are a cornerstone of precision medicine. To advance this modern paradigm of medicine in our ageing society, we propose to employ methods from data science, machine learning, and data visualisations to design and implement a system for computer-aided discovery (CAD) of biomarkers. Current solutions for discovery of prognostic biomarkers from clinical survival data consist of disparate code fragments in R or Python that require a lot of work to be integrated with knowledge bases, and more work still to then be able to interactively and visualy explore the results. There is a great need for comprehensive tools that would assist the domain experts in uncovering hidden patterns and communicate the results to other stakeholders in the process, e.g. clinicians and health regulators.

We propose to develop a set of computational methods and interactive model exploration techniques to democratize the science of biomarker discovery. We will focus on complex multi-gene expression (transcriptomic) biomarkers that are prognostic of cancer patients' clinical outcome (e.g. overal survival, progression free survival). The tool will find and rank potential biomarker genesets, and visualize the relationships between genes in a geneset using ontologies, controlled vocabulary and other forms of curated public knowledge. This will open up the black box of a data-driven approach to biomarker discovery, and make the results interpretable in a wider context of biomedical science. 

The deliverables of the project will empower newcomers and experts from academia and the industry for faster biomarker discovery. We will achieve that by focusing on three aspects. 
1.	Robust Data Science. Developing and open-sourcing a coherent Python library of computational approaches, including survival-based biomarker interaction analysis, biomarker maps, and heuristic search for groups of biomarkers through the integration of data and knowledge-bases.
2.	Ease of Use. Implementing the tools within Orange, an established framework for visual programming and data science, with interactive visualisations that seamlesly integrate with public repositories of data and knowledge.
3.	Communication. The guiding principle for software design will be improved communication between stakeholders in the R&D. But we also aim to empower newcomers to the field with a range of training materials. 

The project is ambitious and challenging, but builds on our previous work. The project leader has published approaches in feature interaction discovery, feature construction, data projection and mapping techniques, knowledge-based search and interpretation of data visualizations. We have been developing Orange (http://orangedatamining.com), which has a huge user base in the industry and in education, and this work will result in a new add-on to the platform. We are partnering with Genialis, an SME specializing in data science research for precision medicine. They are leaders in the area of complex transcriptomic biomarkers and are currently registering the first-ever transcriptomic biomarker for clinical use. The tools developed in this project will primarily speed up their discovery process and simplify the communication with their partners, and a joint patent is envisioned as a direct result of this work. But ultimately, these tools will be made available to a wide scientific community and will have a real potential to globally advance the field of precision medicine.
